awx/main/tasks/host_indirect.py

```diff
@@ -49,18 +49,19 @@ def build_indirect_host_data(job: Job, job_event_queries: dict[str, dict[str, st
         if 'res' not in event.event_data:
             continue
 
-        if 'resolved_action' not in event.event_data or event.event_data['resolved_action'] not in job_event_queries.keys():
+        if not (resolved_action := event.event_data.get('resolved_action', None)):
             continue
 
-        resolved_action = event.event_data['resolved_action']
+        resolved_action_no_module = resolved_action.rsplit('.', 1)[0]
```

Comment:
Should I code more defensively here? Will `resolved_action` ever _not_ be `a.b.c` ?

Reply:
No, not if it comes from our sources (like, runner). But I believe the user could force another value in with a custom module, so for that case I agree you should have some kind of exception handling.